mmcls/models/backbones/gcvit.py

```diff
@@ -0,0 +1,792 @@
+
+    def forward(self, x):
+        b, c, _, _ = x.size()
+        y = self.avg_pool(x).view(b, c)
+        y = self.fc(y).view(b, c, 1, 1)
+        return x * y
+
+
+class ReduceSize(nn.Module):
+    """Down-sampling block based on: "Hatamizadeh et al., Global Context Vision
+    Transformers <https://arxiv.org/abs/2206.09959>"."""
+
+    def __init__(self, dim, norm_layer=nn.LayerNorm, keep_dim=False):
+        """
+        Args:
+            dim: feature size dimension.
+            norm_layer: normalization layer.
+            keep_dim: bool argument for maintaining the resolution.
+        """
+
+        super().__init__()
+        self.conv = nn.Sequential(
+            nn.Conv2d(dim, dim, 3, 1, 1, groups=dim, bias=False),
+            nn.GELU(),
+            SE(dim, dim),
+            nn.Conv2d(dim, dim, 1, 1, 0, bias=False),
+        )
+        if keep_dim:
+            dim_out = dim
+        else:
+            dim_out = 2 * dim
+        self.reduction = nn.Conv2d(dim, dim_out, 3, 2, 1, bias=False)
+        self.norm2 = norm_layer(dim_out)
+        self.norm1 = norm_layer(dim)
+
+    def forward(self, x):
+        x = x.contiguous()
+        x = self.norm1(x)
+        x = _to_channel_first(x)
+        x = x + self.conv(x)
+        x = self.reduction(x)
+        x = _to_channel_last(x)
+        x = self.norm2(x)
+        return x
+
+
+class PatchEmbed(nn.Module):
+    """Patch embedding block based on: "Hatamizadeh et al., Global Context
+    Vision Transformers <https://arxiv.org/abs/2206.09959>"."""
+
+    def __init__(self, in_chans=3, dim=96):
+        """
+        Args:
+            in_chans: number of input channels.
+            dim: feature size dimension.
+        """
+
+        super().__init__()
+        self.proj = nn.Conv2d(in_chans, dim, 3, 2, 1)
+        self.conv_down = ReduceSize(dim=dim, keep_dim=True)
+
+    def forward(self, x):
+        x = self.proj(x)
+        x = _to_channel_last(x)
+        x = self.conv_down(x)
+        return x
+
+
+class FeatExtract(nn.Module):
+    """Feature extraction block based on: "Hatamizadeh et al., Global Context
+    Vision Transformers <https://arxiv.org/abs/2206.09959>"."""
+
+    def __init__(self, dim, keep_dim=False):
+        """
+        Args:
+            dim: feature size dimension.
+            keep_dim: bool argument for maintaining the resolution.
+        """
+
+        super().__init__()
+        self.conv = nn.Sequential(
+            nn.Conv2d(dim, dim, 3, 1, 1, groups=dim, bias=False),
+            nn.GELU(),
+            SE(dim, dim),
+            nn.Conv2d(dim, dim, 1, 1, 0, bias=False),
+        )
+        if not keep_dim:
+            self.pool = nn.MaxPool2d(kernel_size=3, stride=2, padding=1)
+        self.keep_dim = keep_dim
+
+    def forward(self, x):
+        x = x.contiguous()
+        x = x + self.conv(x)
+        if not self.keep_dim:
+            x = self.pool(x)
+        return x
+
+
+class WindowAttention(nn.Module):
```

Comment:
Maybe we can use https://github.com/open-mmlab/mmclassification/blob/2037260ea6c98a3b115e97727e1151a1c2c32f7a/mmcls/models/utils/attention.py#L16 to replace this module